nautobot/extras/views.py

```diff
@@ -483,6 +487,154 @@ class CustomLinkBulkDeleteView(generic.BulkDeleteView):
     table = tables.CustomLinkTable
 
 
+#
+# Dynamic Groups
+#
+
+
+class DynamicGroupListView(generic.ObjectListView):
+    queryset = DynamicGroup.objects.all()
+    table = tables.DynamicGroupTable
+    filterset = filters.DynamicGroupFilterSet
+    filterset_form = forms.DynamicGroupFilterForm
+    action_buttons = ("add",)
+
+
+class DynamicGroupView(generic.ObjectView):
+    queryset = DynamicGroup.objects.all()
+
+    def get_extra_context(self, request, instance):
+        context = super().get_extra_context(request, instance)
+        model = instance.content_type.model_class()
+        table_class = get_table_for_model(model)
+
+        if table_class is not None:
+            members = instance.get_queryset()
+            members_table = table_class(members, orderable=False)
+            context["members_table"] = members_table
+
+        return context
+
+
+class DynamicGroupEditView(generic.ObjectEditView):
+    queryset = DynamicGroup.objects.all()
+    model_form = forms.DynamicGroupForm
+    template_name = "extras/dynamicgroup_edit.html"
+
+    def get_extra_context(self, request, instance):
+        ctx = super().get_extra_context(request, instance)
+
+        filterform_class = instance.generate_filter_form()
+
+        if filterform_class is None:
+            # FIXME(jathan): There is currently an edge case here that needs to be addressed:
+            # `AttributeError: 'NoneType' object has no attribute 'is_valid'`
+            # See: https://sentry.io/share/issue/fb41c6afb40248f6931021574bc38a0d/
```

Comment:
This edge case may have been alleviated, but there is still a valid case where `instance.generate_filter_form()` will return `None`, such as when a `DynamicGroup` object is new and unsaved.